src/cli/index.js

```diff
@@ -339,7 +339,9 @@ function onUnexpectedError(err: Error) {
   log.push(`Trace: ${indent(err.stack)}`);
 
   const errorLoc = path.join(config.cwd, 'yarn-error.log');
-  fs.writeFileSync(errorLoc, log.join('\n\n') + '\n');
+  try {
+    fs.writeFileSync(errorLoc, log.join('\n\n') + '\n');
+  } catch (err) {}
```

Comment:
This should probably at least `console.error` or something rather than silently ignoring it.